### PR TITLE
Guard EraE/eth_subscribe overflow, drop pre-ulong leftovers, cover gas-limit reject arm

### PR DESCRIPTION
**Pull request by wurdum**
Follow-up hardening and cleanup for the `long → ulong` type-unification. Three production-relevant edge cases that the unification left exposed are guarded, the dead pre-`ulong` scaffolding it superseded is removed, and the consensus gas-limit cap regains its missing reject-arm test coverage. No behavioral change for valid inputs.

## Changes

**Bugfixes — unsigned overflow/underflow at boundaries**

- **EraE import resume buffer underflow** (`Nethermind.EraE/Import/EraImporter.cs`): the `BlockTreeSuggestPacer` resume batch size was computed as `ImportBlocksBufferSize - 1024UL`, which wraps to ~2^64 when `ImportBlocksBufferSize <= 1024`. Now guarded — subtract `1024` only when the buffer exceeds it, otherwise fall back to `ImportBlocksBufferSize / 2`. Brings EraE in line with the Era1 importer's underflow handling.
- **`eth_subscribe` over-width / malformed params** (`Nethermind.JsonRpc/Modules/Subscribe/SubscribeRpcModule.cs`): the subscription-args parse only caught `JsonException`, so an over-width or negative block parameter (e.g. `fromBlock: "0x10000000000000000"` = 2^64, or `"-1"`) surfaced as an `OverflowException`/`FormatException` and escaped as an InternalError. The catch now also handles `FormatException` and `OverflowException`, mapping these to the standard `InvalidParams` (-32602).

**Refactor — remove leftovers of pre-`ulong` times**

- **EthStats history-range normalization** (`Nethermind.EthStats/Integrations/EthStatsIntegration.cs`): with `min`/`max` now `ulong`, the `if (max < 0)` guard is unreachable and `min = Math.Max(0, min)` is a no-op. Removed both. Since normalization can no longer fail, `TryNormalizeHistoryRange` (returning `bool`) becomes `NormalizeHistoryRange` (returning `void`), and the dead "invalid range" early-return in `SendHistoryAsync` is dropped. Behavior-preserving.
- **Optimism payload-id buffer size** (`Nethermind.Optimism/Rpc/OptimismPayloadAttributes.cs`): `ComputePayloadIdMembersSize` reserved `sizeof(long)` for the now-`ulong` `gasLimit`; changed to `sizeof(ulong)`. Both are 8 bytes — purely cosmetic alignment with the write path, no payload-id change.
- **HeaderValidator gas-limit cap reject arm** (`Nethermind.Blockchain.Test/Validators/HeaderValidatorTests.cs`): the unification replaced the dead signed `< 0` checks with an explicit `GasLimit > 2^63-1` consensus cap (`ValidateFieldLimit`), but only the accept boundary (`== 2^63-1`) was covered. Converted the single accept-only test into a parameterized boundary sweep `When_gas_limit_around_protocol_cap` that also exercises the reject arm at `2^63` and `ulong.MaxValue`, asserting the rejection is attributed to `BlockErrorMessages.InvalidGasLimit`.

#### What types of changes does your code introduce?

- [x] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [x] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [x] Yes
- [ ] No

#### If yes, did you write tests?

- [x] Yes
- [ ] No

#### Requires documentation update

- [ ] Yes
- [x] No

#### Requires explanation in Release Notes

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/Nethermind/Nethermind.EraE/Import/EraImporter.cs [hunk 1]
```diff
@@ -92,7 +92,10 @@ private async Task ImportInternal(ulong from, ulong to, IEraStore eraStore, Canc
         using ProgressReporter progress = new("EraE import", logManager, to - from + 1);
         ulong blocksProcessed = 0;
 
-        using BlockTreeSuggestPacer pacer = new(blockTree, eraConfig.ImportBlocksBufferSize, eraConfig.ImportBlocksBufferSize - 1024UL);
+        ulong resumeBatchSize = eraConfig.ImportBlocksBufferSize > 1024UL
+            ? eraConfig.ImportBlocksBufferSize - 1024UL
```
Comment (Marchhill) [inline src/Nethermind/Nethermind.EraE/Import/EraImporter.cs:96]: add helper and reuse also in era1

**Review by Marchhill — APPROVED** [review]

**Review by LukaszRozmej — APPROVED** [review]

**Review by svlachakis — APPROVED** [review]
I would add a quick regression test (1024 → 512, 1025 → 1). Seems to be the only fix without coverage.

**Review by AnkushinDaniil — APPROVED** [review]

File: src/Nethermind/Nethermind.EraE/Import/EraImporter.cs (hunk 1 above)
Comment (wurdum) [reply src/Nethermind/Nethermind.EraE/Import/EraImporter.cs:96]: I don't see an easy way to unify this logic. `EraE` and `Era1` live in completely independent projects. I don't know why such a design decision was made, but changing it here feels wrong. Putting this logic into `BlockTreeSuggestPacer` also doesn't look good.

I'd rather leave it as is. What do you think?
